source/base/mpi_consensus_algorithm.cc

```diff
@@ -0,0 +1,41 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/mpi_consensus_algorithm.templates.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace Utilities
+{
+  namespace MPI
+  {
+    template class ConsensusAlgorithmProcess<unsigned int, unsigned int>;
+
+    template class ConsensusAlgorithmSelector<unsigned int, unsigned int>;
+
+
+    template class ConsensusAlgorithmProcess<
+      std::pair<types::global_dof_index, types::global_dof_index>,
```

Comment:
I guess it's best to use 
```suggestion
      std::pair<std::uint64_t, std::uint64_t>,
```
to make 32-bit compilations happy and avoid an additional `#ifdef` here.

Reply:
I am not sure I understand, shouldn't global_dof_index be the right thing?

Reply:
My guess the problems are these lines (in an other file):

https://github.com/dealii/dealii/blob/e47f572be1a8b6c46db5ef33dcdc4e1c2f526ed1/source/base/mpi.cc#L1048-L1050

which I have removed in the last push.

Reply:
The problem is that `types::global_dof_index` is `unsigned int` on non-64-bit compilations, so we have duplicate instantiations with the one a few lines above. But I agree that we should be explicit on `types::global_dof_index` and rather put an `#ifdef DEAL_II_WITH_64BIT_INDICES` around the `unsinged int` instantiations.

Reply:
Ah sorry, that's right. We have a pair here anyway so there is no clash with the instantiation above.